src/datepicker/datepicker-input.ts

```diff
@@ -276,9 +280,30 @@ export class NgbInputDatepicker implements OnChanges,
       this._cRef.instance.focus();
 
       // closing on ESC
-      fromEvent<KeyboardEvent>(this._document, 'keyup')
-          .pipe(takeUntil(this._closed$), filter(e => e.which === Key.Escape))
-          .subscribe(() => this.close());
+      // clang-format off
```

Comment:
Please, no `clang-format off`...

Reply:
@pkozlowski-opensource you don't want to maintain observables formatted with clang

Reply:
@benouat I don't want to maintain _any_ formatting. If we go down this path we should remove clang altogether. 

We either use a tool that works or we remove it. IMO being half-way is a worst place to be....